delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ProjectsRoutes.scala

```diff
@@ -100,50 +105,72 @@ final class ProjectsRoutes(
-                          entity(as[ProjectFields]) { fields =>
-                            emit(StatusCodes.Created, projects.create(ref, fields).mapValue(_.metadata))
+              concat(
+                operationName(s"$prefixSegment/projects/{org}/{project}") {
+                  concat(
+                    (put & pathEndOrSingleSlash) {
+                      parameter("rev".as[Long].?) {
+                        case None      =>
+                          // Create project
+                          authorizeFor(ref, projectsPermissions.create).apply {
+                            entity(as[ProjectFields]) { fields =>
+                              emit(StatusCodes.Created, projects.create(ref, fields).mapValue(_.metadata))
+                            }
                           }
-                        }
-                      case Some(rev) =>
-                        // Update project
-                        authorizeFor(ref, projectsPermissions.write).apply {
-                          entity(as[ProjectFields]) { fields =>
-                            emit(projects.update(ref, rev, fields).mapValue(_.metadata))
+                        case Some(rev) =>
+                          // Update project
+                          authorizeFor(ref, projectsPermissions.write).apply {
+                            entity(as[ProjectFields]) { fields =>
+                              emit(projects.update(ref, rev, fields).mapValue(_.metadata))
+                            }
                           }
+                      }
+                    },
+                    (get & pathEndOrSingleSlash) {
+                      authorizeFor(ref, projectsPermissions.read).apply {
+                        parameter("rev".as[Long].?) {
+                          case Some(rev) => // Fetch project at specific revision
+                            emit(projects.fetchAt(ref, rev).leftWiden[ProjectRejection])
+                          case None      => // Fetch project
+                            emit(projects.fetch(ref).leftWiden[ProjectRejection])
                         }
-                    }
-                  },
-                  (get & pathEndOrSingleSlash) {
-                    authorizeFor(ref, projectsPermissions.read).apply {
-                      parameter("rev".as[Long].?) {
-                        case Some(rev) => // Fetch project at specific revision
-                          emit(projects.fetchAt(ref, rev).leftWiden[ProjectRejection])
-                        case None      => // Fetch project
-                          emit(projects.fetch(ref).leftWiden[ProjectRejection])
+                      }
+                    },
+                    // Deprecate/delete project
+                    (delete & pathEndOrSingleSlash) {
+                      parameters("rev".as[Long], "wipe".?(false)) {
+                        case (rev, true) if config.allowResourcesDeletion =>
```

Comment:
I think in this case is better this way. Notice that in case of deletion we have to do extra things (creating the location, and returning a SeeOther).

If we had that on the `Projects` trait it would have to return some weird response type like `IO[ProjectRejection, (Option[UUID], ProjectResource)]`.
So the current solution is more concise than the proposed solution. I don't see a reason why to change it.